lib/manageiq/appliance_console/message_configuration_server.rb

```diff
@@ -0,0 +1,225 @@
+          configure_messaging_yaml     # Set up the local message client in case EVM is actually running on this, Message Server
+        rescue AwesomeSpawn::CommandResultError => e
+          say(e.result.output)
+          say(e.result.error)
+          say("")
+          say("Failed to Configure the Message Server- #{e}")
+          return false
+        rescue => e
+          say("Failed to Configure the Message Server- #{e}")
+          return false
+        end
+        true
+      end
+
+      def post_activation
+        say("Starting zookeeper and configure it to start on reboots ...")
+        LinuxAdmin::Service.new("zookeeper").start.enable
+
+        say("Starting kafka and configure it to start on reboots ...")
+        LinuxAdmin::Service.new("kafka").start.enable
+      end
+
+      def ask_for_parameters
+        say("\nMessage Server Parameters:\n\n")
+
+        @username  = ask_for_string("Message Key Username", @username)
+        @password  = ask_for_password("Message Key Password")
+      end
+
+      def show_parameters
+        say("\nMessage Server Configuration:\n")
+        say("Message Server Details:\n")
+        say("  Message Server Hostname:   #{@server_hostname}\n")
+        say("  Message Key Username:      #{@username}\n")
+      end
+
+      private
+
+      def my_hostname
+        LinuxAdmin::Hosts.new.hostname
+      end
+
+      def create_jaas_config
+        say(__method__.to_s.tr("_", " ").titleize)
+
+        content = <<~JAAS
+          KafkaServer {
+            org.apache.kafka.common.security.plain.PlainLoginModule required
+            username=#{@username}
+            password=#{@password}
+            user_admin=#{@password} ;
+          };
+        JAAS
+
+        File.write(@jaas_config_path, content) unless file_found?(@jaas_config_path)
+      end
+
+      def create_logs_directory
+        say(__method__.to_s.tr("_", " ").titleize)
+
+        return true if file_found?(LOGS_DIR)
+
+        FileUtils.mkdir_p(LOGS_DIR)
+        FileUtils.chmod(0o755, LOGS_DIR)
+        FileUtils.chown("kafka", "kafka", LOGS_DIR)
+      end
+
+      def configure_firewall
+        say(__method__.to_s.tr("_", " ").titleize)
+
+        AwesomeSpawn.run!("firewall-cmd --add-port=9092/tcp --permanent") # insecure
+        AwesomeSpawn.run!("firewall-cmd --add-port=9093/tcp --permanent") # secure
+        AwesomeSpawn.run!("firewall-cmd --reload")
+      end
+
+      def configure_keystore
+        say(__method__.to_s.tr("_", " ").titleize)
+
+        return true if file_found?(@keystore_dir_path)
+
+        FileUtils.mkdir_p(@keystore_dir_path)
+        FileUtils.chmod(0o755, @keystore_dir_path)
+
+        # Generte a Java keystore and key pair, creating keystore.jks
+        AwesomeSpawn.run!("keytool -keystore #{@keystore_path} -alias localhost -validity 10000 -genkey -keyalg RSA -storepass #{@password} -keypass #{@password} -dname \"cn=#{@server_hostname}\" ")
```

Comment:
AwesomeSpawn should be used to handle the individual parameters to ensure they are properly escaped.  For example, if someone puts a `|` character in their password, the current way is broken.  Prefer:

```suggestion
        AwesomeSpawn.run!("keytool", :params => {"-keystore" => @keystore_path, "-alias" => "localhost", "-validity" => 10000, "-genkey" => nil, "-keyalg" => "RSA", "-storepass" => @password, "-keypass" => @password, "-dname" => "cn=#{@server_hostname}")
```

Reply:
@Fryguy Thank you for you input. I've addressed this and the other AwesomeSpawn issue with commit: https://github.com/ManageIQ/manageiq-appliance_console/pull/130/commits/37575cb580cab9d0404b9640c9bad22706ae28e8